Cloud cli update (#188)
OpenLiberty/draft-guide-cloud-ibm: README.adoc

Different cloud-based solutions are available to run your Kubernetes workloads. A cloud-based infrastructure enables you to focus on developing your microservices without worrying about low-level infrastructure details for deployment. Using a cloud helps you to easily scale and manage your microservices in a high-availability setup.

> The IBM Cloud Kubernetes Service is part of IBM’s public cloud offerings. It provides a hosted Kubernetes cluster where you can deploy your microservices. In this guide, you will use it with the IBM Cloud Container Registry, which is a private registry that is used to store and distribute your container images. Please note, the IBM Cloud Container Registry is hosted on IBM Cloud Platform, fees might be associated with running this guide. Please refer to https://cloud.ibm.com/docs/account?topic=account-accounts[IBM Cloud account type] document for more details. 

Reviewer: ```suggestion
The IBM Cloud Kubernetes Service is an IBM public cloud offering. It provides a hosted Kubernetes cluster where you can deploy your microservices. In this guide, you will use it with the IBM Cloud Container Registry, which is a private registry that is used to store and distribute your container images. 

Note: Because the IBM Cloud Container Registry is hosted on IBM Cloud Platform, fees might be associated with running this guide. For more information, see link:https://cloud.ibm.com/docs/account?topic=account-accounts[IBM Cloud account type]. 
```

=== Provisioning a cluster

> To create a {kube} cluster on IBM Cloud, you need `Administrator` access to IBM Cloud Kubernetes Service. To verify your `Administrator` access, log in to the https://cloud.ibm.com/[IBM Cloud Dashboard^]. Then, navigate to `Manage > Access (IAM) > Users > [Your Username] > Access Policies`. Alternatively, if you are part of an access group, you also can check `Manage > Access (IAM) > Access groups > [Your Access Group] > Access`. Using the steps mentioned previously, you can confirm that `Administrator` is listed as a policy for all resources in the account or for the Kubernetes service.

Reviewer 2: ```suggestion
To create a {kube} cluster on IBM Cloud, you need `Administrator` access to IBM Cloud Kubernetes Service. To verify your `Administrator` access, log in to the https://cloud.ibm.com/[IBM Cloud Dashboard^]. Then, navigate to `Manage > Access (IAM) > Users > [Your Username] > Access > Access Policies`. Alternatively, if you are part of an access group, you also can check `Manage > Access (IAM) > Access groups > [Your Access Group] > Access`. Using the steps mentioned previously, you can confirm that `Administrator` is listed as a policy for all resources in the account or for the Kubernetes service.
```

Once you have confirmed that you have appropriate permissions, use the following commands to provision a cluster.

Retrieve the list of available Kubernetes zone for deploying the cluster within the https://cloud.ibm.com/docs/containers?topic=containers-regions-and-zones#zones-sz[IBM Cloud classic infrastructure] 
[role=command]
```
ibmcloud ks zone ls --provider classic
```

> Replace `[available_zone]` with an available zone selected from the previous command. Then use the following command to retrieve the list of the available flavors of Kubernetes worker nodes for a zone.

Reviewer 2: ```suggestion
Use the following command to retrieve the list of the available flavors of Kubernetes worker nodes for a zone by replacing the `[available_zone]` with an available zone selected from the previous command. 
```

[role=command]
```
ibmcloud ks flavor ls  --provider classic --zone [available_zone]
```

You will see the following output: 
[role="no_copy"]
----
For more information about these flavors, see 'https://ibm.biz/flavors'
Name                      Cores   Memory   Network Speed   OS             Server Type   Storage   Secondary Storage   Flavor Class   Provider
b3c.16x64                 16      64GB     1000Mbps        UBUNTU_20_64   virtual       25GB      100GB               -              classic

> b3c.16x64.300gb           16      64GB     1000Mbps        UBUNTU_20_64   virtual       25GB      300GB               -              classic

Reviewer 2: ```suggestion
b3c.16x64.300gb           16      64GB     1000Mbps        UBUNTU_20_64   virtual       25GB      300GB               -              classic
...
```

----

> Replace `[selected_flavor]` with an available flavor selected from the output of the previous command, and replace `[available_zone]` with the previously used zone. Then use the following command to create a Kubernetes cluster.

Reviewer 2: ```suggestion
Use the following command to create a Kubernetes cluster by replacing the `[selected_flavor]` with an available flavor selected from the output of the previous command, and the `[available_zone]` with the previously used zone.
```

=== Provisioning a cluster

> To create a {kube} cluster on IBM Cloud, you need `Administrator` access to IBM Cloud Kubernetes Service. To verify your `Administrator` access, log in to the https://cloud.ibm.com/[IBM Cloud Dashboard^]. Then, navigate to `Manage > Access (IAM) > Users > [Your Username] > Access > Access Policies`. Alternatively, if you are part of an access group, you also can check `Manage > Access (IAM) > Access groups > [Your Access Group] > Access`. Using the steps mentioned previously, you can confirm that `Administrator` is listed as a policy for all resources in the account or for the Kubernetes service.

Reviewer: ```suggestion
To create a {kube} cluster on IBM Cloud, you need Administrator access to IBM Cloud Kubernetes Service. To verify your Administrator access, log in to the https://cloud.ibm.com/[IBM Cloud Dashboard^]. Then, navigate to **Manage** > **Access (IAM**) > **Users** > [Your Username] > **Access** > **Access Policies**`. Alternatively, if you are part of an access group, you also can check `**Manage** > **Access (IAM)** > **Access groups** > [Your Access Group] > **Access**`. Using these steps, you can confirm that `Administrator` is listed as a policy for all resources in the account or for the Kubernetes service.
```

> Once you have confirmed that you have appropriate permissions, use the following commands to provision a cluster.

Reviewer: ```suggestion
Once you confirm that you have appropriate permissions, use the following commands to provision a cluster.
```

> Retrieve the list of available Kubernetes zone for deploying the cluster within the https://cloud.ibm.com/docs/containers?topic=containers-regions-and-zones#zones-sz[IBM Cloud classic infrastructure] 

Reviewer: ```suggestion
Retrieve the list of available Kubernetes zones for deploying the cluster within the https://cloud.ibm.com/docs/containers?topic=containers-regions-and-zones#zones-sz[IBM Cloud classic infrastructure] 
```

[role=command]
```
ibmcloud ks zone ls --provider classic
```

> Use the following command to retrieve the list of the available flavors of Kubernetes worker nodes for a zone by replacing the `[available_zone]` with an available zone selected from the previous command. 

Reviewer: ```suggestion
Use the following command to retrieve a list of the available flavors of Kubernetes worker nodes for a zone. Replace the `[available_zone]` value with a zone selected from the list you generated with previous command. 
```

[role=command]
```
ibmcloud ks flavor ls  --provider classic --zone [available_zone]
```

> You will see the following output: 

Reviewer: ```suggestion
Look for the following output: 
```

[role="no_copy"]
----
For more information about these flavors, see 'https://ibm.biz/flavors'
Name                      Cores   Memory   Network Speed   OS             Server Type   Storage   Secondary Storage   Flavor Class   Provider
b3c.16x64                 16      64GB     1000Mbps        UBUNTU_20_64   virtual       25GB      100GB               -              classic
b3c.16x64.300gb           16      64GB     1000Mbps        UBUNTU_20_64   virtual       25GB      300GB               -              classic
...
----

> Use the following command to create a Kubernetes cluster by replacing the `[selected_flavor]` with an available flavor selected from the output of the previous command, and the `[available_zone]` with the previously used zone.

Reviewer: ```suggestion
Use the following command to create a Kubernetes cluster. Replacing the `[selected_flavor]` value with a flavor selected from the output of the previous command and the `[available_zone]` with the previously used zone.
```

[role=command]
```
ibmcloud ks cluster create classic --name guide-cluster --flavor [selected_flavor] --zone [available_zone]
```

> You will see the following output:

Reviewer: ```suggestion
Look for the following output:
```

Next, run the `docker build` commands to build container images for your application:
[role='command']
```
docker build --platform linux/amd64 -t system:1.0-SNAPSHOT system/.
docker build --platform linux/amd64 -t inventory:1.0-SNAPSHOT inventory/.
```

> The `--platform` flag in the `docker build` command allows user to specify the target platforms for the build output. As IBM Cloud Kubernetes service support both `x86_64` and `s390x` architectures, the target platform will be `linux/amd64`. In addition, the `-t` flag in the command allows the Docker image to be labeled (tagged) in the `name[:tag]` format. The tag for an image describes the specific image version. If the optional `[:tag]` tag is not specified, the `latest` tag is created by default.

Reviewer 2: ```suggestion
The `--platform` flag in the `docker build` command allows user to specify the target platform for the build output. As IBM Cloud Kubernetes service supports both `x86_64` and `s390x` architectures, the target platform will be `linux/amd64`. In addition, the `-t` flag in the command allows the Docker image to be labeled (tagged) in the `name[:tag]` format. The tag for an image describes the specific image version. If the optional `[:tag]` tag is not specified, the `latest` tag is created by default.
```

Next, run the `docker build` commands to build container images for your application:
[role='command']
```
docker build --platform linux/amd64 -t system:1.0-SNAPSHOT system/.
docker build --platform linux/amd64 -t inventory:1.0-SNAPSHOT inventory/.
```

> The `--platform` flag in the `docker build` command allows user to specify the target platform for the build output. As IBM Cloud Kubernetes service supports both `x86_64` and `s390x` architectures, the target platform will be `linux/amd64`. In addition, the `-t` flag in the command allows the Docker image to be labeled (tagged) in the `name[:tag]` format. The tag for an image describes the specific image version. If the optional `[:tag]` tag is not specified, the `latest` tag is created by default.

Reviewer: ```suggestion
The `--platform` flag in the `docker build` command specifies the target platform for the build output. Because IBM Cloud Kubernetes service supports both `x86_64` and `s390x` architectures, the target platform is `linux/amd64`. Also, the `-t` flag in the command allows the Docker image to be labeled (tagged) in the `name[:tag]` format. The tag for an image describes the specific image version. If the optional `[:tag]` tag is not specified, the `latest` tag is created by default.
```

Pushing the images to a registry enables the cluster to create pods by using your container images. It's a private repository so only users with access to your IBM Cloud account will have access to these images.

The registry you will use is called IBM Cloud Container Registry. Use the container registry plug-in to create a namespace for your container images. 

> To create, assign, and remove namespaces, you must have the Manager role in the Container Registry service at the account level. For more information, please visit https://cloud.ibm.com/docs/Registry?topic=Registry-registry_setup_cli_namespace#registry_setup_cli_namespace_plan_perm[User permissions for working with namespaces]

Reviewer 2: ```suggestion
To create, assign, and remove namespaces, you must have the Manager role in the Container Registry service at the account level. For more information, see the https://cloud.ibm.com/docs/Registry?topic=Registry-registry_setup_cli_namespace#registry_setup_cli_namespace_plan_perm[User permissions for working with namespaces] documentation.
```

Pushing the images to a registry enables the cluster to create pods by using your container images. It's a private repository so only users with access to your IBM Cloud account will have access to these images.

The registry you will use is called IBM Cloud Container Registry. Use the container registry plug-in to create a namespace for your container images. 

> To create, assign, and remove namespaces, you must have the Manager role in the Container Registry service at the account level. For more information, see the https://cloud.ibm.com/docs/Registry?topic=Registry-registry_setup_cli_namespace#registry_setup_cli_namespace_plan_perm[User permissions for working with namespaces] documentation.

Reviewer: ```suggestion
To create, assign, and remove namespaces, you must have the Manager role in the Container Registry service at the account level. For more information, see link:https://cloud.ibm.com/docs/Registry?topic=Registry-registry_setup_cli_namespace#registry_setup_cli_namespace_plan_perm[User permissions for working with namespaces].
```
